### PR TITLE
Nd 166 read shapes from osm

### PR DESCRIPTION
**Pull request by nberard**
needs https://github.com/CanalTP/osm-transit-extractor/pull/38 and osm-transit-extractor tagged

### CODE REVIEW

**Comment by nberard** [comment]
jenkins, test this please

File: src/read_shapes.rs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    }
+}
+
+impl WithGeometry for NtfsRoute {
+    fn set_geometry_id(&mut self, geometry_id: String) -> Result<()> {
+        if self.geometry_id.is_some() {
+            bail!("geometry already exists for route {:?}", self.id);
+        } else {
+            self.geometry_id = Some(geometry_id);
+            return Ok(());
```
Comment (datanel) [inline src/read_shapes.rs]: Ok(())

File: src/read_shapes.rs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+{
+    let mut objects = ntfs_objects.take();
+    let mut generated_geo_obj_id = 0;
+    for obj in objects.iter_mut() {
+        if let Some((_, osm_obj_id)) = obj
+            .codes()
+            .iter()
+            .find(|(key, _)| *key == format!("osm_{}_id", object_type))
+        {
+            generated_geo_obj_id = generated_geo_obj_id + 1;
```
Comment (datanel) [inline src/read_shapes.rs]: generated_geo_obj_id += 1;

File: src/read_shapes.rs [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+}
+
+pub trait WithGeometry {
+    fn set_geometry_id(&mut self, geometry_id: String) -> Result<()>;
+}
+
+impl WithGeometry for NtfsLine {
+    fn set_geometry_id(&mut self, geometry_id: String) -> Result<()> {
+        if self.geometry_id.is_some() {
+            bail!("geometry already exists for line {:?}", self.id);
```
Comment (datanel) [inline src/read_shapes.rs]: do we have to return an error or just warn?

Comment (datanel) [reply src/read_shapes.rs]: I would rather do it outside.
```rust
pub trait WithGeometry {
    fn set_geometry_id(&mut self, geometry_id: String);
    fn geometry_id(&self) -> &Option<String>;
}

impl WithGeometry for NtfsLine {
    fn set_geometry_id(&mut self, geometry_id: String) {
        self.geometry_id = Some(geometry_id);
    }

    fn geometry_id(&self) -> &Option<String> {
        &self.geometry_id
    }
}

impl WithGeometry for NtfsRoute {
    fn set_geometry_id(&mut self, geometry_id: String) {
        self.geometry_id = Some(geometry_id);
    }
    fn geometry_id(&self) -> &Option<String> {
        &self.geometry_id
    }
}
...
match obj.geometry_id() {
    Some(geo_id) => obj.set_geometry_id(geo_id.clone()),
    None => warn!("geometry already exists for {} {:?}", object_type, obj.id()),
}
```

**Pull request merged**